packages/mambajs/src/solver.ts

```diff
@@ -43,30 +43,69 @@ export interface ISolveOptions {
 const solve = async (
   specs: Array<string>,
   channels: Array<string>,
+  installedCondaPackages: ISolvedPackages,
   logger?: ILogger
 ) => {
   let result: SolvedPackage[] | undefined = undefined;
   const solvedPackages: ISolvedPackages = {};
   try {
+    let installed: any = [];
+    if (Object.keys(installedCondaPackages).length) {
+      Object.keys(installedCondaPackages).map((filename: string) => {
+        const installedPkg = installedCondaPackages[filename];
+        let tmpPkg = {
+          ...installedPkg,
+          packageName: installedPkg.name,
+          build: installedPkg.build_string,
+          filename
+        };
+
+        installed.push(tmpPkg);
+      });
+    } else {
+      installed = undefined;
+    }
     const startSolveTime = performance.now();
-    result = await simpleSolve(specs, channels, PLATFORMS);
+    result = (await simpleSolve(
+      specs,
+      channels,
+      PLATFORMS,
+      installed
+    )) as SolvedPackage[];
     const endSolveTime = performance.now();
-
     if (logger) {
       logger.log(
         `Solving took ${(endSolveTime - startSolveTime) / 1000} seconds`
       );
     }
-
+    result.forEach((pkg: any) => {
```

Comment:
Same here, we shouldn't shadow the actual type of `item`

```suggestion
    result.forEach((pkg) => {
```